ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -577,6 +577,86 @@ public void deleteErrorCasesTest() {
     verifyDeleteFailure(getUniqueId(), StoreErrorCodes.ID_Not_Found);
   }
 
+  /**
+   * Test DELETE authorization success for {@link BlobStore#delete(MessageWriteSet)}.
+   * If either accountId or containerId is -1, validation is skipped.
+   */
+  @Test
+  public void deleteAuthorizationSuccessTest() throws Exception {
+    short[] accountIds = {-1, Utils.getRandomShort(TestUtils.RANDOM), -1};
+    short[] containerIds = {-1, -1, Utils.getRandomShort(TestUtils.RANDOM)};
+    for (int i = 0; i < accountIds.length; i++) {
+      MockId mockId = putSingleBlob(UtilsTest.getRandomString(10), accountIds[i], containerIds[i]);
+      MessageInfo info = new MessageInfo(mockId, DELETE_RECORD_SIZE, Utils.getRandomShort(TestUtils.RANDOM),
+          Utils.getRandomShort(TestUtils.RANDOM), System.currentTimeMillis());
+      MessageWriteSet writeSet = new MockMessageWriteSet(Collections.singletonList(info),
+          Collections.singletonList(ByteBuffer.allocate(DELETE_RECORD_SIZE)));
+      store.delete(writeSet);
+      verifyDeleteFailure(mockId, StoreErrorCodes.ID_Deleted);
+    }
+  }
+
+  /**
+   * Test DELETE authorization failure for {@link BlobStore#delete(MessageWriteSet)}.
+   */
+  @Test
+  public void deleteAuthorizationFailureTest() throws Exception {
+    if (store.getStoreConfig().storeValidateAuthorization == true) {
```

Comment:
you can use `assumeTrue(store.getStoreConfig().storeValidateAuthorization)` - the test won't run if it is not true (will get skipped, won't fail).

Reply:
nice.